unipop-core/src/org/unipop/structure/UniVertexProperty.java

```diff
@@ -1,17 +1,19 @@
 package org.unipop.structure;
 
+import org.apache.commons.lang.NotImplementedException;
 import org.apache.tinkerpop.gremlin.structure.*;
 import org.apache.tinkerpop.gremlin.structure.util.*;
 
-import java.util.Iterator;
+import java.util.*;
 
-public class UniVertexProperty<V> implements VertexProperty<V> {
+public class UniVertexProperty<V> extends UniElement implements VertexProperty<V> {
```

Comment:
I'm not so hot on the idea of every property containing:
```
    protected HashMap<String, Property> properties = new HashMap<>();
    protected String id;
    protected String label;
```
I'm afraid that it could bump up the memory. If its not too much trouble I'm for reverting this back until we tackle meta-properties.